app/src/components/tables/Pagination/NumberedPagination.tsx

```diff
@@ -0,0 +1,306 @@
+import styled from '@emotion/styled';
+import { ActionCreatorWithPayload } from '@reduxjs/toolkit';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { SelectOptions } from 'src/components/layout/Header/Partials';
+import { CustomSelect } from 'src/components/layout/Header/Partials/CustomSelect';
+import { useAppDispatch } from 'src/store';
+import { TableOptions } from 'src/store/types';
+import { colors, fontWeight, pxToRem } from 'src/styled-theme';
+import { loadConfig, standardizeNumber } from 'src/utils';
+import lessThanWhite from '../../../assets/images/less-than-white.png';
+import moreThanWhite from '../../../assets/images/more-than-white.png';
+
+const { defaultPagination } = loadConfig();
+
+interface NumberedPaginationProps {
+  rowCountSelectOptions: SelectOptions[];
+  tableOptions: TableOptions;
+  setTableOptions: ActionCreatorWithPayload<TableOptions, string>;
+  setIsTableLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  totalPages: number;
+  updatePageNum: ActionCreatorWithPayload<number, string>;
+}
+
+export const NumberedPagination: React.FC<NumberedPaginationProps> = ({
+  rowCountSelectOptions,
+  tableOptions,
+  setTableOptions,
+  setIsTableLoading,
+  totalPages,
+  updatePageNum,
+}) => {
+  const { t } = useTranslation();
+
+  const dispatch = useAppDispatch();
+
+  const defaultRowCountOption = rowCountSelectOptions[1];
+
+  const [currentRowCountOption, setCurrentRowCountOption] = useState(
+    defaultRowCountOption,
+  );
+
+  useEffect(() => {
+    dispatch(
+      setTableOptions({
+        ...tableOptions,
+        pagination: {
+          ...tableOptions.pagination,
+          pageSize: Number(currentRowCountOption.value) ?? defaultPagination,
+        },
+      }),
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentRowCountOption]);
```

Comment:
we don't need dispatch, tableOptions of defaultPagination in here hey?

Reply:
Actually, I added back `dispatch` - it was `tableOptions` and `defaultPagination` that I didn't want here.